### PR TITLE
added pagination logic

### PR DESCRIPTION
**Pull request by IK-kiranpipaliya**
**JIRA**

 - https://uplevel-watch.atlassian.net/jira/software/c/projects/UG/boards/34?selectedIssue=UG-2390 
 

**Descriptions**
  -  Implemented `createPagination()` function to dynamically generate pagination controls.
  - The function accepts two parameters: `containerSelector` for targeting the pagination container and `defaultParam` to set a default page key if it's not present in the URL.
  - Pagination buttons are generated based on the current page and the total number of pages retrieved from the DOM.
  - Includes support for previous and next navigation buttons, along with ellipses for skipped page ranges.
  - Automatically handles active page highlighting and ensures edge cases, such as when there are fewer pages than the maximum visible pages, are covered.
  - Example usage provided with event listener to trigger pagination creation on page load.
  
  
image

### CODE REVIEW

File: pagination-logic/pagination.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/**
+ * Creates a pagination control dynamically.
+ *
+ * @param {string} containerSelector - The CSS selector for the pagination container element.
+ * @param {string} defaultParam - The default query parameter key to be used if the page key is not found in the URL.
+ */
+function createPagination(containerSelector, defaultParam) {
```
Comment (KDholakiya) [inline pagination-logic/pagination.js:7]: Where are you calling this function?

**Review by KDholakiya — COMMENTED** [review]
where are we going to link these files?

File: pagination-logic/pagination.js (hunk 1 above)
Comment (IK-kiranpipaliya) [reply pagination-logic/pagination.js:7]: after importing this script to our project we can call this in our webflow site like this 
`document.addEventListener('DOMContentLoaded', createPagination('.parent-class', urlDefaultParam));`

**Comment by IK-kiranpipaliya** [comment]
> where are we going to link these files?

we can add this script link to body tag and use function and same we add css link to our head tag

**Review by KDholakiya — APPROVED** [review]

**Pull request merged**